units/src/fee_rate/serde.rs

```diff
@@ -0,0 +1,254 @@
+
+    pub fn serialize<S: Serializer>(f: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+        u64::serialize(&f.to_sat_per_kwu(), s)
+    }
+
+    pub fn deserialize<'d, D: Deserializer<'d>>(d: D) -> Result<FeeRate, D::Error> {
+        Ok(FeeRate::from_sat_per_kwu(u64::deserialize(d)?))
+    }
+
+    pub mod opt {
+        //! Serialize and deserialize [`Option<FeeRate>`] denominated in satoshis per 100 weight units.
+        //!
+        //! Use with `#[serde(with = "fee_rate::serde::as_sat_per_kwu::opt")]`.
+
+        use core::fmt;
+
+        use serde::{de, Deserialize, Deserializer, Serializer};
+
+        use crate::FeeRate;
+
+        pub fn serialize<S: Serializer>(f: &Option<FeeRate>, s: S) -> Result<S::Ok, S::Error> {
+            match *f {
+                Some(f) => s.serialize_some(&f.to_sat_per_kwu()),
+                None => s.serialize_none(),
+            }
+        }
+
+        pub fn deserialize<'d, D: Deserializer<'d>>(d: D) -> Result<Option<FeeRate>, D::Error> {
+            struct VisitOpt;
+
+            impl<'de> de::Visitor<'de> for VisitOpt {
+                type Value = Option<FeeRate>;
+
+                fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    write!(f, "An Option<FeeRate>")
+                }
+
+                fn visit_none<E>(self) -> Result<Self::Value, E>
+                where
+                    E: de::Error,
+                {
+                    Ok(None)
+                }
+
+                fn visit_some<D>(self, d: D) -> Result<Self::Value, D::Error>
+                where
+                    D: Deserializer<'de>,
+                {
+                    Ok(Some(FeeRate::from_sat_per_kwu(u64::deserialize(d)?)))
+                }
+            }
+            d.deserialize_option(VisitOpt)
+        }
+    }
+}
+
+pub mod as_sat_per_vb_floor {
+    //! Serialize and deserialize [`FeeRate`] denominated in satoshis per virtual byte.
+    //!
+    //! When serializing use floor division to convert per kwu to per virtual byte.
+    //! Use with `#[serde(with = "fee_rate::serde::as_sat_per_vb_floor")]`.
+
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    use crate::fee_rate::serde::OverflowError;
+    use crate::fee_rate::FeeRate;
+
+    pub fn serialize<S: Serializer>(f: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+        u64::serialize(&f.to_sat_per_vb_floor(), s)
+    }
+
+    // Errors on overflow.
+    pub fn deserialize<'d, D: Deserializer<'d>>(d: D) -> Result<FeeRate, D::Error> {
+        Ok(FeeRate::from_sat_per_vb(u64::deserialize(d)?)
+            .ok_or(OverflowError)
+            .map_err(serde::de::Error::custom)?)
+    }
+
+    pub mod opt {
+        //! Serialize and deserialize [`Option<FeeRate>`] denominated in satoshis per virtual byte.
+        //!
+        //! When serializing use floor division to convert per kwu to per virtual byte.
+        //! Use with `#[serde(with = "fee_rate::serde::as_sat_per_vb_floor::opt")]`.
```

Comment:
This comment is the exact same as on L91 - L94 except for `::opt`.  This should probably describe more why `opt` is different than mod `as_sat_per_vb_floor`.

Reply:
Thanks. I'm leaving it as is.